doozerlib/runtime.py

```diff
@@ -344,6 +348,32 @@ def filter_disabled(n, d):
                 if not self.image_map:
                     self.logger.warning("No image metadata directories found for given options within: {}".format(self.group_dir))
 
+                for image in self.image_map.itervalues():
+                    image.resolve_parent()
+
+                self.image_tree = {}
+                image_lists = {0: []}
+
+                def add_child_branch(child, branch, level=1):
```

Comment:
Another time I'm not 100% sure why this has a be a nested function. Is that simply your preference for local stuff you don't think needs to live in the top level namespace for modules? I think if you expanded that argument list you could include a reference to the `image_lists` object.

Reply:
It's a personal style thing for stuff that's recursive. I don't believe a function should exist in any scope it's not used.